### PR TITLE
cmake: install public headers

### PR DESCRIPTION
**Pull request by SpaceIm**
closes https://github.com/pytorch/tensorpipe/issues/317

I understood from https://github.com/pytorch/tensorpipe/issues/317#issuecomment-799329872 that there is more generally a need to track all headers, but it's already a good improvement to track public headers.

### CODE REVIEW

**Review by lw — COMMENTED** [review]
Awesome! Thanks so much for taking care of this!

I only have one request: by building the list of headers at the very end, we're replicating some of the "logic" that we already have in place above (I'm talking about the `if(...)` and such). By having that occur twice, far apart, we have the risk of forgetting to update one of them after we update the other. I think it would be safer to build the list of headers as we go, in order to have a unique logic that builds both the source files and the headers. Therefore, could you move the new code to the appropriate locations (the core list of headers after the core list of source files, the headers for channel X after the sources for channel X, and so on...)? Thanks!

**Comment by SpaceIm** [comment]
something like this?

File: cmake/Options.cmake [hunk 1]
```diff
@@ -48,3 +48,5 @@ option(TP_BUILD_LIBUV "Build libuv from source" OFF)
 include(GNUInstallDirs)
 set(TP_INSTALL_LIBDIR ${CMAKE_INSTALL_LIBDIR} CACHE STRING "Directory in which to install libraries")
 mark_as_advanced(TP_INSTALL_LIBDIR)
+set(TP_INSTALL_INCLUDEDIR ${CMAKE_INSTALL_INCLUDEDIR} CACHE STRING "Directory in which to install public headers")
+mark_as_advanced(TP_INSTALL_INCLUDEDIR)
```
Comment (SpaceIm) [inline cmake/Options.cmake:52]: Is it something you want to keep? It's quite unusual.

Comment (lw) [reply cmake/Options.cmake:52]: I must admit I don't remember why we have it. Is there harm in keeping it? What's the "preferred" way?

**Review by lw — APPROVED** [review]
This looks great! Thanks again!

Let me know if I can go ahead and merge it or if you want to iterate more on it.

File: cmake/Options.cmake (hunk 1 above)
Comment (SpaceIm) [reply cmake/Options.cmake:52]: It doesn't harm I think. If consumer wants to change the standard install layout (bin, lib, include, share folders), I guess it's useful, but uncommon.

**Comment by SpaceIm** [comment]
@lw, no more iteration for me, you can merge if you want ;)

**Pull request closed without merging**